scrapper.py

```diff
@@ -21,43 +32,189 @@ class IncorrectNumberOfArticlesError(Exception):
     """
 
 
+class HTMLParser:
+    """
+    Parser implementation
+    """
+    def __init__(self, article_url, article_id):
+        """
+        Init
+        """
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(url=article_url, article_id=article_id)
+
+    def parse(self):
+        """
+
+        """
+        response = requests.get(self.article_url)
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        self.article.save_raw()
+
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+        """
+        Fills self.article with text from article_bs
+        """
+
+        table_rows = article_bs.find_all('tr', {"class": "unnrow"})
+
+        for table_row in table_rows:
+            table_datas = table_row.find_all('td')
+
+            if not table_datas:
+                continue
+
+            if "Загрузить статью" in table_datas[0].get_text():
+                pdf_url = table_datas[1].find('a')
+
+                pdf_raw_file = PDFRawFile(pdf_url['href'], self.article_id)
+
+                pdf_raw_file.download()
+                text = pdf_raw_file.get_text()
+
+                parts_of_article = text.split('Список литературы')
+
+                self.article.text = ''.join(parts_of_article[:-1])
+                return
+
+    def _fill_article_with_meta_information(self, article_bs):
+        """
+        Fills self.article with meta information
+        """
+
+        self.article.title = article_bs.find('h3').get_text()
+
+        tables = article_bs.find_all('table', {"class": "unntable"})
+
+        for table in tables:
+            table_datas = table.find_all('td')
+
+            if table_datas[0].get_text() != 'Авторы':
+                continue
+
+            link_to_author = table_datas[1].find('a')
+
+            self.article.author = link_to_author.get_text()
+
+        text_date = re.search(r'Поступила в редакцию\s+\d{2}\.\d{2}\.\d{4}', self.article.text)
+
+        if text_date:
```

Comment:
if not we could create a date from the volume number given the regularity of volume publishes

Reply:
It is always present there